File: apps/auto-wechat/frontend/src/types/artifacts.ts
import type { PipelineStep, StepStatus } from '@/constants/pipeline'

export interface RankedItem {
  url: string
  title: string
  score: number
  reason?: string
  summary?: string
  summaryZh?: string
  tags?: string[]
  source?: string
}

export interface DigestArtifact {
  id: string
  runId: string
  items: RankedItem[]
  stats?: Record<string, unknown>
}

export interface ContentDraftArtifact {
  id: string
  runId: string
  digestId?: string
  title: string
  summary: string
  bodyMarkdown: string
  bodyHtml: string
  coverUrl?: string
  coverMediaId?: string
  readSourcePresetId?: string
  status: string
  editor?: Record<string, unknown>
  review?: Record<string, unknown>
}

export interface StepDetailArtifact {
  step: PipelineStep
  status: StepStatus
  input?: unknown
  output?: unknown
  errorMessage: string | null
  durationMs?: number | null
  startedAt: string | null
  finishedAt: string | null
}

export interface PublishResultArtifact {
  draftMediaId: string
  publishMode: string
  createdAt?: string
}

export interface RunArtifacts {
  runId: string
  runStatus: string
  publishMode: string
  digest?: DigestArtifact
  contentDraft?: ContentDraftArtifact
  steps: StepDetailArtifact[]
  draftMediaId?: string | null
  previewHtml?: string | null
  publishResult?: PublishResultArtifact
}

export interface UpdateDraftPayload {
  title?: string
  summary?: string
  bodyMarkdown?: string
  bodyHtml?: string
  coverUrl?: string
  readSourcePresetId?: string
}

export interface PublishRunPayload {
  readSourcePresetId?: string
}

export interface UpdateDigestPayload {
  items: RankedItem[]
}

/** POST /pipeline/runs/:id/steps/:step/regenerate 受理回执（非实时进度） */
export interface RegenerateStepResponse {
  runId: string
  step: PipelineStep
  status: 'running'
  affectedSteps: PipelineStep[]
  cascade: boolean
}

File: apps/auto-wechat/backend/internal/domain/artifacts.go
package domain

import (
	"encoding/json"
	"time"
)

type StepDetail struct {
	Step         PipelineStep
	Status       StepStatus
	InputJSON    json.RawMessage
	OutputJSON   json.RawMessage
	ErrorMessage *string
	DurationMs   *int
	StartedAt    *time.Time
	FinishedAt   *time.Time
}

type RunArtifacts struct {
	RunID         string
	RunStatus     RunStatus
	PublishMode   string
	Digest        *Digest
	ContentDraft  *ContentDraft
	Steps         []StepDetail
	DraftMediaID  *string
	PreviewHTML   *string
	PublishResult *PublishRecord
}

type PublishRecord struct {
	DraftMediaID string
	PublishMode  string
	CreatedAt    time.Time
}
